cli/src/core/parse/validate.go

```diff
@@ -0,0 +1,91 @@
+package parse
+
+import (
+	"path"
+	"strings"
+
+	"cli/core"
+	"cli/util/slice"
+
+	"github.com/luno/jettison/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	ErrNoConfigImage            = errors.New("config file missing field 'image'")
+	ErrNoPackages               = errors.New("no packages or custom packages specified")
+	ErrUndefinedProfilePackages = errors.New("packages in profile not in any of packages, custom-packages, or command-line custom-packages")
+	ErrNoSuchProfile            = errors.New("no such profile")
+)
+
+func validate(cmd *cobra.Command, config *core.Config) error {
+	customPackagePaths, err := cmd.Flags().GetStringSlice("custom-path")
+	if err != nil {
+		return errors.Wrap(err, "")
+	}
+
+	if config.Image == "" {
+		return errors.Wrap(ErrNoConfigImage, "")
+	} else if len(config.Packages) == 0 && len(config.CustomPackages) == 0 && len(customPackagePaths) == 0 {
+		return errors.Wrap(ErrNoPackages, "")
+	}
+
+	profileName, err := cmd.Flags().GetString("profile")
+	if err != nil {
+		return errors.Wrap(err, "")
+	}
+
+	if profileName != "" {
+		return validateProfile(cmd, profileName, config)
+	}
+
+	return nil
+}
+
+func validateProfile(cmd *cobra.Command, profileName string, config *core.Config) error {
+	profilePackagesMap := make(map[string]bool)
+	var profile core.Profile
+	for _, prof := range config.Profiles {
+		if prof.Name == profileName {
+			for _, p := range prof.Packages {
+				profilePackagesMap[p] = true
+				profile = prof
+			}
+			
+			break
+		}
+	}
+
+	if len(profilePackagesMap) < 1 {
+		return errors.Wrap(ErrNoSuchProfile, profileName)
```

Comment:
I think this is a misleading error, the profile can exist with no packages in it
so maybe we can split two cases : 
- No such profile means the profile name does not exist in config file
- No packages defined in profile name 

Or we can edit the error message to be: no such profile or the profiles defined has no packages specified

Reply:
Yes, you are entirely right! Fixed this in [f3a068c](https://github.com/openhie/package-starter-kit/pull/36/commits/f3a068cd8e75a7c5f8ebe34b5e58ad8ae8e525c6)